src/utils/ReactDOMMini.js

```diff
@@ -0,0 +1,168 @@
+}
+
+function escapeTextForBrowser(text) {
+  if (typeof text === 'boolean' || typeof text === 'number') {
+    return '' + text;
+  }
+  return escapeHtml(text.trim());
+}
+
+const reconciler = ReactReconciler({
+  supportsMutation: true,
+  // eslint-disable-next-line no-unused-vars
+  finalizeInitialChildren(element, type, props) {},
+  // eslint-disable-next-line no-unused-vars
+  getChildHostContext(parentContext, fiberType, rootInstance) {},
+  // eslint-disable-next-line no-unused-vars
+  getRootHostContext(rootInstance) {},
+  // eslint-disable-next-line no-unused-vars
+  shouldSetTextContent(type, props) {},
+  createTextInstance(
+    text,
+    // eslint-disable-next-line no-unused-vars
+    rootContainerInstance,
+    // eslint-disable-next-line no-unused-vars
+    hostContext,
+    // eslint-disable-next-line no-unused-vars
+    internalInstanceHandle,
+  ) {
+    return text;
+  },
+  // eslint-disable-next-line no-unused-vars
+  createInstance(type, props, rootContainerInstance, hostContext) {
+    const { children, dangerouslySetInnerHTML, ...rest } = props;
+    const res = {
+      tagName: type,
+      attributes: rest,
+    };
+    Object.keys(res.attributes).forEach((key) => {
+      if (res.attributes[key] === undefined) delete res.attributes[key];
+    });
+    if (!type.startsWith('mj')) {
+      return {
+        type,
+        props,
+        children: [],
+        isReact: true,
+      };
+    }
+
+    if (props.dangerouslySetInnerHTML && props.dangerouslySetInnerHTML.__html) {
+      res.content = escapeTextForBrowser(props.dangerouslySetInnerHTML.__html);
+    }
+    return res;
+  },
+  clearContainer() {},
+  appendChildToContainer(container, child) {
+    container.resultObj = child;
+  },
+  // eslint-disable-next-line no-unused-vars
+  appendChild(parent, child) {},
+  appendInitialChild(parent, child) {
+    if (child.isReact) {
+      if (parent.isReact) {
+        parent.children.push(child);
+      } else {
+        const reactElement = toReactElement(child);
+        if (!parent.content) {
+          parent.content = '';
+        }
+        parent.content += escapeTextForBrowser(
+          ReactDOMServer.renderToStaticMarkup(reactElement),
+        );
+      }
+    } else if (typeof child === 'string') {
+      if (!child) return;
+      if (parent.isReact) {
+        parent.children.push(child);
+      } else {
+        if (!parent.content) {
+          parent.content = '';
+        }
+        parent.content += escapeTextForBrowser(child);
+      }
+    } else {
+      if (!parent.children) parent.children = [];
+      parent.children.push(child);
+    }
+  },
+  // eslint-disable-next-line no-unused-vars
+  prepareForCommit(...args) {},
+  // eslint-disable-next-line no-unused-vars
+  resetAfterCommit(...args) {},
+});
+
+const ReactDOMMini = {
+  render(whatToRender) {
+    const container = reconciler.createContainer({}, false, false);
```

Comment:
On each `render` we are creating a new container. Does `reconciler` have some method for stopping/shutdown/cleanup ? Try running a `10000nth` times and check memory usage if we don't have memory leak.